peps/pep-0738.rst

```diff
@@ -0,0 +1,363 @@
+
+Unlike iOS, creating subprocesses *is* supported on Android. However apps may
+only run executables in `certain locations
+<https://issuetracker.google.com/issues/128554619#comment4>`__, none of which
+are writable at runtime. Long-running subprocesses are `officially discouraged
+<https://issuetracker.google.com/issues/128554619#comment4>`__, and are not
+guaranteed to be supported in future Android versions.
+
+Android does provide a command-line shell, but this is intended only for use by
+developers, and is not available to the typical end user.
+
+For these reasons, the primary way of running Python on Android will be by
+loading ``libpython3.x.so`` into the main app process. Although there will also
+be a ``python3.x`` executable linked against ``libpython3.x.so``, this is only
+for debugging, not production use.
+
+
+Specification
+=============
+
+Scope of work
+-------------
+
+The focus of this work will be to produce an Android equivalent to the existing
+Windows embeddable package, i.e. a set of compiled libraries which developers
+can add to their apps. No installer will be required.
+
+Adding Android as a Tier 3 platform only requires adding support for compiling
+an Android-compatible build from the unpatched CPython source code. It does not
+necessarily require there to be any officially distributed Android artefacts on
+python.org, although these could be added in the future.
+
+CPython will not recommend any particular way of packaging Python into an app
+and accessing it over JNI. The details of this are left to higher-level tools
+(see `How to Teach This`_).
+
+A Gradle project will be provided for the purpose of running the CPython test
+suite. Tooling will be provided to automate the process of building the test
+suite app, starting the emulator, installing the test suite, and executing
+it.
+
+
+Linkage
+-------
+
+For the reasons discussed in `App lifecycle`_, Python will be included in the
+app as a dynamic ``libpython3.x.so`` library. Android extension modules should
+all be linked against this library. This allows using the
+``-Wl,--no-undefined`` option to detect missing symbols at build time, which can
+be a significant time-saver.
+
+An extension module linked against ``libpython3.x.so`` cannot be loaded by an
+executable that has been statically linked against ``libpython3.x.a``.
+Therefore, a static ``libpython3.x.a`` library will not be supported on Android.
+This is the same pattern used by CPython on Windows.
+
+Unlike iOS, Android allows dynamic libraries to be loaded from any location, so
+a directory tree containing co-located .py, .pyc and .so files can be handled by
+Python's standard importer.
+
+
+Standard library
+----------------
+
+A number of standard library modules will not be supported on Android because
+the underlying C APIs are not provided by the operating system:
+
+* ``curses`` and ``readline``
+* ``dbm.gnu`` and ``dbm.ndbm``
+* ``grp``, ``nis`` and ``spwd``
+
```

Comment:
The paragraph below doesn't make it clear whether `multiprocessing` will be omitted, or made to work.
If it'll be omitted, I suggest making the list exhaustive – that'll make things clear and more skimmable:
```suggestion
* ``multiprocessing``, and ``tkinter``, ``turtle`` (see below)

```

Reply:
I've clarified that it will be omitted, and merged everything into a single list.